dask_kubernetes/operator/daskcluster.py

```diff
@@ -111,3 +167,78 @@ async def daskcluster_create(spec, name, namespace, logger, **kwargs):
         f"A scheduler service has been created called {data['metadata']['name']} in {namespace} \
         with the following config: {data['spec']}"
     )
+
+    data = build_worker_group_spec("default", spec.get("image"), spec.get("workers"))
+    kopf.adopt(data)
+    api = kubernetes.client.CustomObjectsApi()
+    worker_pods = api.create_namespaced_custom_object(
+        group="kubernetes.dask.org",
+        version="v1",
+        plural="daskworkergroups",
+        namespace=namespace,
+        body=data,
+    )
+    logger.info(
+        f"A worker group has been created called {data['metadata']['name']} in {namespace} \
+        with the following config: {data['spec']}"
+    )
+
+
+@kopf.timer("daskworkergroup", interval=5.0)
+@kopf.on.create("daskworkergroup")
+async def daskworkergroup_create(spec, name, namespace, logger, **kwargs):
+    api = kubernetes.client.CoreV1Api()
+
+    scheduler = kubernetes.client.CustomObjectsApi().list_cluster_custom_object(
+        group="kubernetes.dask.org", version="v1", plural="daskclusters"
+    )
+    scheduler_name = scheduler["items"][0]["metadata"]["name"]
+    scheduler_spec = scheduler["items"][0]["spec"]
+    scheduler_data = build_scheduler_pod_spec(
+        name=scheduler_name, image=scheduler_spec.get("image")
+    )
+    kopf.adopt(scheduler_data)
+    workers = api.list_namespaced_pod(
+        namespace=namespace,
+        label_selector=f"dask.org/workergroup-name={name}",
+    )
+    current_workers = len(workers.items)
+    desired_workers = spec["workers"]["replicas"]
+    workers_needed = desired_workers - current_workers
+    if workers_needed > 0:
+        for i in range(current_workers + 1, desired_workers + 1):
+            data = build_worker_pod_spec(
+                name, namespace, spec.get("image"), i, scheduler_name
+            )
+            kopf.adopt(data)
+            worker_pod = api.create_namespaced_pod(
+                namespace=namespace,
+                body=data,
+            )
+        logger.info(f"{workers_needed} Worker pods in created in {namespace}")
+    if workers_needed < 0:
+        pass
+
+
+@kopf.timer("daskworkergroup", interval=5.0)
+async def scale_workers(spec, name, namespace, logger, **kwargs):
```

Comment:
I'm not really sure what this is doing.

Reply:
This patching/changing a `DaskWorkerGroup` resource with new data/spec. When the `DaskWorkerGroup` is patched, the `daskworkergroup_create` function will update the number of workers to reflect the new spec.